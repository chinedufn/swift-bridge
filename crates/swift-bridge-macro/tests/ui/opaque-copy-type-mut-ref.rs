//! # To Run
//! cargo test -p swift-bridge-macro -- ui trybuild=opaque-copy-type-mut-ref.rs

// We declare a function that returns a mutable Copy type and a method takes takes a mutable
// reference to a Copy type.
//
// We do not support bridge Copy types mutably since the bytes are copied over the bridge.
#[swift_bridge::bridge]
mod ffi {
    extern "Rust" {
        #[swift_bridge(Copy(1))]
        type SomeType;

        fn some_method(&mut self);
        #[swift_bridge(rust_name = "some_method")]
        fn some_method2(self: &mut SomeType);

        fn some_function(arg: &mut SomeType);
    }
}

#[derive(Copy, Clone)]
pub struct SomeType(u8);

impl SomeType {
    fn some_method(&mut self) {}
}
fn some_function(_arg: &mut SomeType) {}

fn main() {}
