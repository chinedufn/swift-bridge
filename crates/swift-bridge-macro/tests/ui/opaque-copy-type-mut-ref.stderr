error: Mutable references to opaque Copy types are not yet supported.
  --> tests/ui/opaque-copy-type-mut-ref.rs:14:24
   |
14 |         fn some_method(&mut self);
   |                        ^^^^^^^^^

error: Mutable references to opaque Copy types are not yet supported.
  --> tests/ui/opaque-copy-type-mut-ref.rs:16:25
   |
16 |         fn some_method2(self: &mut SomeType);
   |                         ^^^^^^^^^^^^^^^^^^^

error: Mutable references to opaque Copy types are not yet supported.
  --> tests/ui/opaque-copy-type-mut-ref.rs:18:26
   |
18 |         fn some_function(arg: &mut SomeType);
   |                          ^^^^^^^^^^^^^^^^^^
